### PR TITLE
LibWeb: Make it easier to parse CSS dimension values

### PR DESCRIPTION
**Pull request by AtkinsSJ**
Frequently we want to parse "anything that's a `<length-percentage>`" or similar, which could be a constant value or a calculation, but didn't have a nice way of doing so. That meant repeating the same "try and parse a dimension, see if it's the right type, then maybe try and parse a calculation and see if that's the right type" boilerplate code. Or more often, forgetting about calculations entirely.

These helpers should make that a lot more convenient to do that. Most of them are currently unused, but I'd rather have them all here now to make it easier to parse things in the future, once they are needed.

The price is an unfortunate amount of boilerplate for all these `parse_foo()` and `parse_foo_percentage()` methods.

### CODE REVIEW

File: Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp [hunk 1]
```diff
@@ -2115,6 +2123,15 @@ void CalculatedStyleValue::CalculationResult::add_or_subtract_internal(SumOperat
                     m_value = Length::make_px(this_px - other_px);
             }
         },
+        [&](Resolution const& time) {
+            auto this_dots_per_pixel = time.to_dots_per_pixel();
```
Comment (Lubrsi) [inline Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp:2127]:
```suggestion
        [&](Resolution const& resolution) {
            auto this_dots_per_pixel = resolution.to_dots_per_pixel();
```

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 2]
```diff
@@ -1832,19 +1832,15 @@ Optional<TimePercentage> Parser::parse_time_percentage(TokenStream<ComponentValu
...
-        return LengthOrCalculated { calculated_value.release_nonnull() };
-    }
-
-    if (auto length = parse_length(component_value); length.has_value()) {
-        return LengthOrCalculated { length.release_value() };
-    }
+    if (auto length = parse_length(tokens); length.has_value())
+        return length;
 
     return {};
```
Comment (Lubrsi) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:1845]: Could this be consolidated to:
```suggestion
    return parse_length(tokens);
```

**Review by Lubrsi — COMMENTED** [review]
Looking good (though I'm not too well versed in the CSS parser myself), just a couple of nits

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp (hunk 2 above)
Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:1845]: You're right! Originally I had a transaction in there but removed it, and forgot to simplify this.

**Comment by AtkinsSJ** [comment]
Changes: Fixed a couple of issues Lubrsi found.

**Pull request merged**